modules/openid_auth/app/controllers/openid_auth/v0/validation_controller.rb

```diff
@@ -1,18 +1,27 @@
 # frozen_string_literal: true
 
 require_dependency 'openid_auth/application_controller'
+require 'mvi/errors/errors'
 
 module OpenidAuth
   module V0
     class ValidationController < ApplicationController
       def index
+        raise StandardError unless @current_user.icn
```

Comment:
This pattern (raising a StandardError and then catching it and re-throwing a more specific error) feels brittle. What about a method like `validate_user` (or a better name) that checks all the possible conditions we want and raises the right error for each one, otherwise returns silently, and then if it returns silently just render the validated_payload?

Reply:
I tried looking for a MVI specific method but couldn't find one except what we use to return the ICN which was @current_user.loa3? but that didn't seem like the right method call. My reason for not going with validate_user was because the payload can validate but still not be connected to MVI or have an ICN which is why when things broke Liberty was still getting a valid response but with a null ICN value, I was trying to basically avoid that so that if something is not connecting to instead return a proper error for the missing ICN connection. I tried using mvi/errors/errors but nothing seemed to be the right error response.